src/favorites/favorites.service.ts

```diff
@@ -0,0 +1,110 @@
+
+@Injectable()
+export class FavoritesService {
+  constructor(
+    private readonly favoritesRepository: FavoritesRepository,
+    private readonly albumService: AlbumsService,
+    private readonly artistService: ArtistsService,
+    private readonly trackService: TracksService,
+  ) {}
+
+  async getAll(): Promise<IFavoritesResponse> {
+    const favorites = this.favoritesRepository.getAll();
+    const entities = await Promise.allSettled([
+      ...favorites.albums.map((id) => this.albumService.getById(id)),
+      ...favorites.artists.map((id) => this.artistService.getById(id)),
+      ...favorites.tracks.map((id) => this.trackService.getById(id)),
+    ]);
+
+    const result: IFavoritesResponse = {
+      albums: [],
+      artists: [],
+      tracks: [],
+    };
+
+    entities.forEach((el) => {
+      if (el.status !== 'fulfilled') {
+        return;
+      }
+      if (el.value instanceof Album) {
+        result.albums.push(el.value);
+      } else if (el.value instanceof Artist) {
+        result.artists.push(el.value);
+      } else if (el.value instanceof Track) {
+        result.tracks.push(el.value);
+      }
+    });
+
+    return result;
+  }
+
+  async addAlbum(id: string): Promise<void> {
+    try {
+      await this.albumService.getById(id);
+      return this.favoritesRepository.addAlbum(id);
+    } catch (e) {
+      console.error(e);
+      throw new UnprocessableEntityException(errorMessages.ALBUM_NOT_FOUND);
+    }
+  }
+
+  async removeAlbum(id: string): Promise<void> {
+    try {
+      await this.albumService.getById(id);
+      return this.favoritesRepository.removeAlbum(id);
+    } catch (e) {
+      console.error(e);
+      throw new UnprocessableEntityException(errorMessages.ALBUM_NOT_FOUND);
+    }
+  }
+
+  async addArtist(id: string): Promise<void> {
+    try {
+      await this.artistService.getById(id);
+      return this.favoritesRepository.addArtist(id);
+    } catch (e) {
+      console.error(e);
+      throw new UnprocessableEntityException(errorMessages.ARTIST_NOT_FOUND);
+    }
+  }
+
+  async removeArtist(id: string): Promise<void> {
+    try {
+      await this.artistService.getById(id);
+      return this.favoritesRepository.removeArtist(id);
+    } catch (e) {
+      console.error(e);
+      throw new UnprocessableEntityException(errorMessages.ARTIST_NOT_FOUND);
+    }
+  }
+
+  async addTrack(id: string): Promise<void> {
+    try {
+      await this.trackService.getById(id);
+      return this.favoritesRepository.addTrack(id);
+    } catch (e) {
+      console.error(e);
+      throw new UnprocessableEntityException(errorMessages.TRACK_NOT_FOUND);
+    }
+  }
+
+  async removeTrack(id: string): Promise<void> {
+    try {
+      await this.trackService.getById(id);
+      return this.favoritesRepository.removeTrack(id);
+    } catch (e) {
+      console.error(e);
+      throw new UnprocessableEntityException(errorMessages.TRACK_NOT_FOUND);
+    }
+  }
+}
```

Comment:
Please show error message if there is no instance in favorites while deletion from favorites.